src/GDGUkraine/rest_controller.py

```diff
@@ -511,6 +511,23 @@ def generate_invites(self, id):
             raise HTTPError(500, "Cannot save generated invites") from e
         return {"ok": True}
 
+    @cherrypy.tools.json_out()
+    @cherrypy.tools.authorize()
+    def record_visit(self, id):
+        '''POST /api/events/:id/check-in'''
+        req = cherrypy.request
+        orm_session = req.orm_session
+        reg_id = int(id)
+        reg_data = api.get_event_registration_by_id(orm_session, reg_id)
+        if not reg_data:
+            raise HTTPError(400,
```

Comment:
Actually (3) is realistic is case of hacking attempts.
Still, I agree w/ @anxolerd, 404 is more verbose. On the other hand, I wouldn't expose the reason of bad request. Let's consider changing this in other dev iteration.